src/main/java/org/folio/circulation/resources/RenewalResource.java

```diff
@@ -73,14 +75,29 @@ private void renew(RoutingContext routingContext) {
       .thenCompose(r -> r.combineAfter(configurationRepository::findTimeZoneConfiguration,
         LoanAndRelatedRecords::withTimeZone))
       .thenComposeAsync(r -> r.after(records -> renewalStrategy.renew(records, bodyAsJson, clients)))
-      .thenComposeAsync(r -> r.after(loanRepository::updateLoan))
+      .thenComposeAsync(r -> r.after(relatedRecords -> updateLoanAndItemInStorage(relatedRecords,
+        itemRepository, loanRepository)))
       .thenApply(r -> r.next(scheduledNoticeService::rescheduleDueDateNotices))
       .thenApply(r -> r.next(loanNoticeSender::sendRenewalPatronNotice))
       .thenApply(r -> r.map(loanRepresentation::extendedLoan))
       .thenApply(LoanResponse::from)
       .thenAccept(result -> result.writeTo(routingContext.response()));
   }
 
+  private CompletableFuture<Result<LoanAndRelatedRecords>> updateLoanAndItemInStorage(
+    LoanAndRelatedRecords relatedRecords, ItemRepository itemRepository,
+    LoanRepository loanRepository) {
+
+    Loan loan = relatedRecords.getLoan();
+    if (loan == null || loan.getItem() == null) {
+      return null;
```

Comment:
Flow will fail with NPE in this case. Can it be converted to something?:
* completableFuture(succeeded(relatedRecords));
* completableFuture(succeeded(null));
* completableFuture(failed("Loan or item is null")).